Source/ASDisplayNode+Layout.mm

```diff
@@ -663,6 +663,10 @@ - (void)transitionLayoutWithSizeRange:(ASSizeRange)constrainedSize
         _pendingLayoutTransition = pendingLayoutTransition = [[ASLayoutTransition alloc] initWithNode:self
                                                                                         pendingLayout:pendingLayout
                                                                                        previousLayout:previousLayout];
+        if ([ASDisplayNode shouldStoreUnflattenedLayouts]) {
```

Comment:
One concern is that `newLayout` used [a few lines above](https://github.com/TextureGroup/Texture/pull/1156/files#diff-71ea742a29543cb7e08c3dffe2a2508eR656) can be unflattened, and it was set as the calculated layout. That's why we needed to flatten it in [the setter](https://github.com/TextureGroup/Texture/pull/1156/files#diff-71ea742a29543cb7e08c3dffe2a2508eR1039).

I think a better approach would be to just flatten that layout immediately and use it throughout this method. Like this

```
{
        // Grab __instanceLock__ here to make sure this transition isn't invalidated
        // right after it passed the validation test and before it proceeds
        ASDN::MutexLocker l(__instanceLock__);

        if ([ASDisplayNode shouldStoreUnflattenedLayouts]) {
          _unflattenedLayout = newLayout;
          newLayout = [newLayout filteredNodeLayoutTree];
        }
        
        // Update calculated layout
        let previousLayout = _calculatedDisplayNodeLayout;
        let pendingLayout = ASDisplayNodeLayout(newLayout,
                                                constrainedSize,
                                                constrainedSize.max,
                                                newLayoutVersion);
        [self _locked_setCalculatedDisplayNodeLayout:pendingLayout];
        
        // Setup pending layout transition for animation
        _pendingLayoutTransition = pendingLayoutTransition = [[ASLayoutTransition alloc] initWithNode:self
                                                                                        pendingLayout:pendingLayout
                                                                                       previousLayout:previousLayout];
        // Setup context for pending layout transition. we need to hold a strong reference to the context
        _pendingLayoutTransitionContext = pendingLayoutTransitionContext = [[_ASTransitionContext alloc] initWithAnimation:animated
                                                                                                            layoutDelegate:_pendingLayoutTransition
                                                                                                        completionDelegate:self];
      }
```